ansible_runner/streaming.py

```diff
@@ -0,0 +1,216 @@
+import codecs
+import io
+import json
+import os
+import stat
+import sys
+import tempfile
+import uuid
+import zipfile
+
+import ansible_runner
+import ansible_runner.plugins
+from ansible_runner import utils
+
+
+class UUIDEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, uuid.UUID):
+            return obj.hex
+        return json.JSONEncoder.default(self, obj)
+
+
+# List of kwargs options to the run method that should be sent to the remote executor.
+remote_run_options = (
+    'forks',
+    'host_pattern',
+    'ident',
+    'ignore_logging',
+    'inventory',
+    'limit',
+    'module',
+    'module_args',
+    'omit_event_data',
+    'only_failed_event_data',
+    'playbook',
+    'verbosity',
+)
+
+
+class Transmitter(object):
+    def __init__(self, _output=None, **kwargs):
+        if _output is None:
+            _output = sys.stdout.buffer
+        self._output = _output
+        self.kwargs = kwargs
+
+        self.status = "unstarted"
+        self.rc = None
+
+    def run(self):
+        remote_options = {key: value for key, value in self.kwargs.items() if key in remote_run_options}
+
+        private_data_dir = self.kwargs.get('private_data_dir', None)
+        self._output.write(
+            utils.stream_dir(private_data_dir, kwargs=json.dumps(remote_options, cls=UUIDEncoder)))
```

Comment:
to be clear, this writes a file named `kwargs` into the private data dir.

That seems like something I'd like to see rolled into the ansible-runner interface contract to start with. I might be able to help with that.

Reply:
Yeah, that's a holdover from the old receptor integration.  I'm thinking it should just be sent in a new initial message type.

Reply:
I'm now sending the kwargs as a separate message to be parsed by `Worker`.